pycloudlib/cloud.py

```diff
@@ -223,3 +223,14 @@ def _streams_query(filters, daily=True):
             raise ValueError(f"No images found matching filters: {filters}")
 
         return result
+
+    def check_and_set_config(self, config_file):
+        """Set pycloudlib configuration.
+
+        Args:
+            config_file: path to pycloudlib configuration file
+        """
+        if config_file:
+            self.config = parse_config(config_file)[self._type]
+        else:
+            self.config = {}
```

Comment:
If `config_file == None` in `BaseCloud.__init__`, we are not trying to read the config from the default locations [CONFIG_PATHS](https://github.com/canonical/pycloudlib/blob/main/pycloudlib/config.py#L11). So, if we try to instantiate an OCI class with or without the new `config_dict` and we have a valid pycloudlib `config` in a default location, we get the following exception:

```python
>>> from pycloudlib import oci
>>> oci.cloud.OCI("test")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/aciba/canonical/pycloudlib/pycloudlib/oci/cloud.py", line 56, in __init__
    availability_domain or self.config["availability_domain"]
KeyError: 'availability_domain'
```

This is also true for other clouds as, this happens in the base class initialization of every cloud.

I think we have to try to read the config in the default locations for other clouds than `OCI` and for `OCI` if `availability_domain`, `compartment_id` or `config_dict` are not given at initialization time.